crates/hotshot/src/traits/networking/libp2p_network.rs

```diff
@@ -528,49 +589,39 @@ impl<M: NetworkMsg, K: SignatureKey + 'static> Libp2pNetwork<M, K> {
         let handle = self.clone();
         let is_bootstrapped = self.inner.is_bootstrapped.clone();
         async_spawn(async move {
-            while let Ok(msg) = handle.inner.handle.receiver().recv().await {
-                match msg {
-                    GossipMsg(msg, _topic) => {
-                        let result: Result<M, _> = bincode_opts().deserialize(&msg);
-                        if let Ok(result) = result {
-                            broadcast_send
-                                .send(result)
-                                .await
-                                .map_err(|_| NetworkError::ChannelSend)?;
-                        }
+            while let Ok(message) = handle.inner.handle.receiver().recv().await {
+                let message_version = match &message {
+                    GossipMsg(raw, _) | DirectRequest(raw, _, _) | DirectResponse(raw, _) => {
+                        read_version(raw)
                     }
-                    DirectRequest(msg, _pid, chan) => {
-                        let result: Result<M, _> = bincode_opts()
-                            .deserialize(&msg)
-                            .context(FailedToSerializeSnafu);
-                        if let Ok(result) = result {
-                            direct_send
-                                .send(result)
-                                .await
-                                .map_err(|_| NetworkError::ChannelSend)?;
-                        }
-                        if handle
-                            .inner
-                            .handle
-                            .direct_response(chan, &Empty::Empty)
-                            .await
-                            .is_err()
-                        {
-                            error!("failed to ack!");
-                        };
+                    NetworkEvent::IsBootstrapped => Some(VERSION_0_1),
```

Comment:
Should we add the version number to the `NetworkEvent::IsBootstrapped` variant, as we do for other variants? Otherwise, it looks like we'll handle other message types differently according to the version number but always use `0.1` for `IsBootstrapped`.

Reply:
I think `IsBootstrapped` is just an internal signal, so I'm not sure about adding a version number (but as you point out, it's just as strange to give it a version number here).

I'll try to reorganize to handle this event separately, without assigning it a version number? Since the only thing we do (as far as I can tell) is set `is_bootstrapped` in `Inner`.